chiselFrontend/src/main/scala/chisel3/core/BlackBox.scala

```diff
@@ -80,7 +80,7 @@ abstract class ExtModule(val params: Map[String, Param] = Map.empty[String, Para
     component
   }
 
-  private[core] def initializeInParent() {
+  private[core] def initializeInParent(instanceCompileOptions: CompileOptions): Unit = {
     implicit val sourceInfo = UnlocatableSourceInfo
 
     for (x <- getModulePorts) {
```

Comment:
Is invalidating black boxes necessary?

Reply:
I'm not sure, I didn't change any of the BlackBox behavior in this PR, we should discuss for 3.1 though